### PR TITLE
[theme] add editor custom styles

### PR DESCRIPTION
**Pull request by tamrat-bay**

### CODE REVIEW

File: packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss [hunk 1]
```diff
@@ -29,7 +29,10 @@
 }
 
 .ricosTiptapEditor {
-  height: 100%;
+  min-height: var(--ricos-custom-editor-min-height, '100%');
+  max-height: var(--ricos-custom-editor-max-height, '100%');
+  overflow-y: auto;
+  padding: 0 20px;
```
Comment (oded-soffrin) [inline packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss:35]: This style need to be applied only when using maxHeight

File: packages/toolbars-v3/web/src/components/FloatingToolbar/FloatingToolbar.tsx [hunk 2]
```diff
@@ -34,8 +34,10 @@ export const FloatingToolbar = ({
   const [dummyUpdate, setForceUpdate] = useState<number>(1);
   const [isModalOpen, setModalOpen] = useState<boolean>(false);
 
+  const domElement = editor.view.dom.parentElement?.parentElement;
```
Comment (oded-soffrin) [inline packages/toolbars-v3/web/src/components/FloatingToolbar/FloatingToolbar.tsx:37]: the maxHeight is implemented on the parent of the parent of the editor dom,
need to either move the css attributes down to contentEditable, or find a better way to get the dom element here

**Review by oded-soffrin — CHANGES_REQUESTED** [review]
some comments above
also - footer toolbar padding is 60px which looks bad in this case. is this by design?

File: packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss (hunk 1 above)
Comment (tamrat-bay) [inline packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss:35]: @lxgreen @oded-soffrin  @danarish4 
wouldn't it result in double padding for consumers who already add padding?

Comment (lxgreen) [reply packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss:35]: @tamrat-bay this is just a quick check. we definitely need to add these rules only if min/maxHeight is applied.

Comment (tamrat-bay) [reply packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss:35]: How about we use JS to check for the custom maxHeight style 
and then use that to apply the rest of the CSS.
I've used it locally and it works fine

Comment (lxgreen) [reply packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss:35]: agree, we can check the theme and apply certain CSS classes. I'd say it should be responsibility of Styles aggregate (`ricos-styles`)

Comment (tamrat-bay) [reply packages/tiptap-editor/web/src/statics/styles/tiptap-editor-styles.scss]: @lxgreen @oded-soffrin 
I have pushed my changes, let me know your thoughts

File: examples/main/src/ExampleApp.tsx [hunk 3]
```diff
@@ -279,13 +279,15 @@ class ExampleApp extends PureComponent<ExampleAppProps, ExampleAppState> {
 
   getThemeObj = (activeFont, activePalette) => {
     const themeObj: RicosTheme = {
+      customStyles: {
+        editor: { minHeight: '200px', maxHeight: '200px' },
```
Comment (lxgreen) [inline examples/main/src/ExampleApp.tsx:283]: TODO: remove this style before merge

File: examples/main/src/ExampleApp.tsx [hunk 4]
```diff
@@ -279,13 +279,15 @@ class ExampleApp extends PureComponent<ExampleAppProps, ExampleAppState> {
...
+      },
       settingsStyles: {
         modals: {
           baseZIndex: 1000,
         },
       },
     };
```
Comment (lxgreen) [inline examples/main/src/ExampleApp.tsx:290]: TODO: revert activeFont

File: packages/tiptap-editor/web/src/components/RicosTiptapEditor/RicosTiptapEditor.tsx [hunk 5]
```diff
@@ -64,6 +64,11 @@ export const RicosTiptapEditor: FunctionComponent<RicosTiptapEditorProps> = ({
 
   const { containerClassName, editorClassName } = editorStyleClasses || {};
 
+  const hasCustomEditorHeight = () => {
+    const styles = window.getComputedStyle(document.body);
+    return styles.getPropertyValue('--ricos-custom-editor-max-height');
+  };
```
Comment (lxgreen) [inline packages/tiptap-editor/web/src/components/RicosTiptapEditor/RicosTiptapEditor.tsx:70]: this should be `Styles` [aggregate](https://github.com/wix/ricos/blob/master/packages/ricos-styles/web/src/styles.ts#L32) responsibility

**Review by lxgreen — CHANGES_REQUESTED** [review]

File: packages/tiptap-editor/web/src/components/RicosTiptapEditor/RicosTiptapEditor.tsx (hunk 5 above)
Comment (oded-soffrin) [reply packages/tiptap-editor/web/src/components/RicosTiptapEditor/RicosTiptapEditor.tsx:70]: regardless of ricos-styles, there's no point in setting the css and than checking via css
we get the theme as JS object we can check the config directly
in addition, `window` doesn't exist on SSR

maybe it's better to sit on it today if we're all in the office ?

File: packages/toolbars-v3/web/src/components/FloatingToolbar/FloatingToolbar.tsx (hunk 2 above)
Comment (tamrat-bay) [reply packages/toolbars-v3/web/src/components/FloatingToolbar/FloatingToolbar.tsx:37]: we can only move the css attributes on level down to `editor.view.dom.parentElement`

File: examples/main/shared/editor/Editor.tsx [hunk 6]
```diff
@@ -184,7 +184,7 @@ export default class Editor extends PureComponent<ExampleEditorProps> {
             content={contentState}
             injectedContent={injectedContent}
             locale={locale}
-            cssOverride={theme}
+            // cssOverride={theme}
```
Comment (tamrat-bay) [inline examples/main/shared/editor/Editor.tsx:187]: TODO: revert before merge

File: packages/toolbars-v3/web/src/components/FloatingToolbar/FloatingToolbar.tsx (hunk 2 above)
Comment (tamrat-bay) [reply packages/toolbars-v3/web/src/components/FloatingToolbar/FloatingToolbar.tsx:37]: @lxgreen 

@oded-soffrin  and I discussed this and decided to leave it this way for now
as well, I am working on improving the solution using refs in the future

File: packages/common/web/statics/styles/palette.scss [hunk 7]
```diff
@@ -47,6 +47,8 @@ $ricos-code-block-font-size: var(--ricos-custom-code-block-font-size, 16px);
 $ricos-code-block-line-height: var(--ricos-custom-code-block-line-height, 26px);
 $ricos-code-block-padding: var(--ricos-custom-code-block-padding, 2px 25px);
 $ricos-code-block-margin: var(--ricos-custom-code-block-margin, 15px 18px);
+$ricos-custom-editor-min-height: var(--ricos-custom-editor-min-height, '100%');
```
Comment (tamrat-bay) [inline packages/common/web/statics/styles/palette.scss:50]: remove fallback 100%

File: packages/ricos-editor/web/src/utils/createEditorStyleClasses.ts [hunk 8]
```diff
@@ -16,12 +18,14 @@ export const createEditorStyleClasses: CreateEditorStyleClasses = ({
...
   editorCss = {},
   experiments = {},
+  theme = {},
 }) => {
   const themeDesktopStyle = cssOverride.desktop
     ? { [cssOverride.desktop]: !isMobile && cssOverride && cssOverride.desktop }
     : {};
   const containerClassName = classNames(editorCss.wrapper, cssOverride.wrapper, {
     [editorCss.desktop]: !isMobile,
+    [editorCss.custom_editor]: theme?.customStyles?.editor?.maxHeight,
```
Comment (tamrat-bay) [inline packages/ricos-editor/web/src/utils/createEditorStyleClasses.ts:28]: refactor - make it easier to read

**Pull request merged**